pkg/virt-controller/watch/migration.go

```diff
@@ -402,36 +413,50 @@ func (c *MigrationController) updateStatus(migration *virtv1.VirtualMachineInsta
 	} else if vmi == nil {
 		migrationCopy.Status.Phase = virtv1.MigrationFailed
 		c.recorder.Eventf(migration, k8sv1.EventTypeWarning, FailedMigrationReason, "Migration failed because vmi does not exist.")
+		decPendingOrRunningMetric(migration, vmi, pod)
+		metrics.IncFailedMigrations(vmi, pod)
```

Comment:
Since these two lines always come together, I'd suggest wrapping them in a function.
What I would do is rename `decPendingOrRunningMetric` to something like `handleFailedMigrationMetrics()` and change the logic to do what both `decPendingOrRunningMetric` and `metrics.IncFailedMigrations` are doing right now.

Something like:
```go
func handleFailedMigrationMetrics(migration *virtv1.VirtualMachineInstanceMigration, vmi *virtv1.VirtualMachineInstance, pod *k8sv1.Pod) {
    if phase := &migration.Status.Phase; phase != nil {
    	if *phase == virtv1.MigrationPending {
    		metrics.DecPendingMigrations(vmi, pod)
    	} else if *phase == virtv1.MigrationRunning {
    		metrics.DecRunningMigrations(vmi, pod)
    	}
    } 
    metrics.IncFailedMigrations(vmi, pod)
}
```

Reply:
done